### PR TITLE
improve doc

### PR DESCRIPTION
**Pull request by FlyingCanoe**
a fix for #47.

I also found what I think is a typo (the UTF-89 in line 440).

### CODE REVIEW

File: src/utf8.rs [hunk 1]
```diff
@@ -43,7 +43,8 @@ const ACCEPT: usize = 12;
 const REJECT: usize = 0;
 
 #[cfg_attr(rustfmt, rustfmt::skip)]
-static CLASSES: [u8; 256] = [
+//SAFETY: unsafe code in the decode function rely on the correctness of this state machine.
```
Comment (BurntSushi) [inline src/utf8.rs]: Could you make this comment (and the ones below) are formatted in a way that is consistent with other comments? Namely:

* Lines should be no longer than 79 columns, inclusive.
* Comments on items like this should use three slashes.
* There should be a space after the `///` and before the first letter.

i.e., `/// Foo` and not `//Foo`.

File: src/utf8.rs [hunk 2]
```diff
@@ -784,6 +786,8 @@ pub fn decode_last_lossy<B: AsRef<[u8]>>(slice: B) -> (char, usize) {
 }
 
 #[inline]
+// SAFETY: the decode function relies on state bean equal to ACCEPT only
+// if cp is a valid Unicode scalar value
```
Comment (BurntSushi) [inline src/utf8.rs]: This comment should go above the `#[inline]` attribute.

Comment (BurntSushi) [reply src/utf8.rs]: Could you use proper English here, as much as possible? Capitalization and ending sentences with a period.

Comment (BurntSushi) [reply src/utf8.rs]: `s/bean/being`

**Comment by FlyingCanoe** [comment]
I think it should be ready. There is a possibility that I have not found some typo in my modification because English is my secondary language.

**Pull request closed without merging**